### DISCUSSION TITLE
Default variables / re-parsing when variables change

### DISCUSSION THREAD

**Discussion by marvinhinz** [deployphp/deployer#2556] [Help needed]
Hi,

for our deployments we use various internal recipes. Often we include multiple recipes in the deployfile. 
Using add("shared_files",[...]) works fine, but i am not sure how to solve the following problem:

Imagine the following setup:

recipe/test.php:
```
set("some_folder", "foo")
add("shared_dirs", [
    "{{some_folder}}/var",
]);

```

deploy.php
```
require "recipe/test.php"
set("some_folder", "bar")
[tasks etc ...]

```
Now when running the deployfile, the shared dirs still contain foo instead of bar. 

How could this test recipe be reused with other variables? In scss for example its possible to declare a variable as `!default`. That means its value will be set only if its not defined yet. Maybe parsing of the strings can be lazyloaded until all global variables are defined?

**Comment by antonmedv** [comment]
Please show the full example. But I suspect currently this is not possible, as "refreshing" no public.
